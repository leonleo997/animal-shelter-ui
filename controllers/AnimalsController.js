import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            url: `animal`,
            baseURL: process.env.API,
            data: animal,
            method: 'POST'
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(animalName) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${animalName}`
        });
    }
}

