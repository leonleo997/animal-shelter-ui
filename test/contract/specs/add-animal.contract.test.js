import { getProvider } from "./init-pact";
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";
import {somethingLike} from "@pact-foundation/pact/src/dsl/matchers";

const provider = getProvider();
describe('Given an Animal Service', () => {
    describe('When a request to create a new animal is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'create animal',
                uponReceiving: 'a request to create an animal',
                withRequest: {
                    method: 'POST',
                    path: '/animal'
                },
                willRespondWith: {
                    status: 201,
                    body: Matchers.somethingLike(
                        {
                            breed: Matchers.like("Bengali"),
                            gender: Matchers.like("Female"),
                            name: Matchers.string("Manchas"),
                            vaccinated: Matchers.boolean(true),
                            // maybe vaccines
                        }
                    )
                }
            });
        });

        it("Then it should return the right data", async() => {
            const animalBody = {name: 'My cat', breed: 'Azul Ruso', gender: 'Male', isVaccinated: true, 'vaccines': ['moquillo', 'gripe', 'leucemia']};
            const response = await AnimalController.register(animalBody)
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        });
    });
});
