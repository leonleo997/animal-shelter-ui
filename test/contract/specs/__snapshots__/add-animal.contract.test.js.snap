// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Animal Service When a request to create a new animal is made Then it should return the right data 1`] = `
Object {
  "breed": "Bengali",
  "gender": "Female",
  "name": "Manchas",
  "vaccinated": true,
}
`;
